client/session.go

```diff
@@ -954,7 +954,7 @@ func (s *session) fetchAllAttempt(
 				successIters := results[:success]
 				resultsLock.RUnlock()
 				iter := s.seriesIteratorPool.Get()
-				iter.Reset(ids[idx], startInclusive, endExclusive, successIters)
+				iter.Reset(tsID.String(), startInclusive, endExclusive, successIters)
```

Comment:
This is quite expensive to cast back to string each time.  Better to either copy the ID and give the series a ts.ID itself, or do ref counting with the ts.ID provided by caller.